### PR TITLE
AW-77 Include basic Chosen example

### PR DESCRIPTION
**Pull request by spg**

### CODE REVIEW

**Comment by Pacane** [comment]
lgtm

**Comment by jasonlemay** [comment]
LGTM

File: pom.xml [hunk 1]
```diff
@@ -28,6 +28,8 @@
         <gsss.version>1.0-SNAPSHOT</gsss.version>
         <gwtquery.version>1.4.2</gwtquery.version>
         <gwt-maps-api.version>3.10.0-alpha-7</gwt-maps-api.version>
+        <gwtchosen.version>2.0.0-SNAPSHOT</gwtchosen.version>
+        <guava.version>17.0</guava.version>
```
Comment (meriouma) [inline pom.xml]: 18.0?

File: pom.xml [hunk 2]
```diff
@@ -92,6 +97,24 @@
             <artifactId>gwt-maps-api</artifactId>
             <version>${gwt-maps-api.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava-gwt</artifactId>
+            <version>${guava.version}</version>
+            <scope>compile</scope>
```
Comment (meriouma) [inline pom.xml]: Any reason you need to explicitly set scope to compile?

File: src/main/java/com/arcbees/website/client/application/products/chosen/ChosenView.java [hunk 3]
```diff
@@ -16,18 +16,36 @@
...
             Binder binder) {
         initWidget(binder.createAndBindUi(this));
+
+        asWidget().addAttachHandler(this);
+    }
+
+    @Override
+    public void onAttachOrDetach(AttachEvent event) {
+        if (event.isAttached()) {
+            $(selectToEnhance).as(Chosen.Chosen).chosen();
```
Comment (meriouma) [inline src/main/java/com/arcbees/website/client/application/products/chosen/ChosenView.java]: You need to use the attach event even if it's an Element? I thought it was only when selecting widgets. I might be confusing things though :P

**Comment by meriouma** [comment]
LGTM

File: src/main/java/com/arcbees/website/client/application/products/chosen/ChosenView.java [hunk 4]
```diff
@@ -16,18 +16,36 @@
...
 
+    @UiField
+    SelectElement selectToEnhance;
+
     @Inject
     ChosenView(
             Binder binder) {
         initWidget(binder.createAndBindUi(this));
+
+        asWidget().addAttachHandler(this);
```
Comment (christiangoudreau) [inline src/main/java/com/arcbees/website/client/application/products/chosen/ChosenView.java]: Using latest GWTP, you can remove this and use onAttach and onDetach overridable methods.

File: pom.xml (hunk 2 above)
Comment (spg) [inline pom.xml]: Nope, will set to `provided`

**Pull request merged**